NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,568 @@
+
+
+def baseline_correction(wfs, n_bins=128, func=np.median):
+   """
+   Simple baseline correction function. Determines baseline in discrete chuncks of "n_bins" with
+   the function specified (i.e., mean or median).
+   
+   Parameters
+   ----------
+   
+   wfs: np.array(n_events, n_channels, n_samples)
+      Waveforms of several events/channels.
+      
+   n_bins: int
+      Number of samples/bins in one "chunck". If None, calculate median/mean over entire trace. (Default: 128)
+      
+   func: np.mean or np.median
+      Function to calculate pedestal
+   
+   Returns
+   -------
+   
+   wfs_corrected: np.array(n_events, n_channels, n_samples)
+      Baseline/pedestal corrected waveforms
+   """
+    
+   # Example: Get baselines in chunks of 128 bins
+   # wfs in (n_events, n_channels, 2048)
+   # np.split -> (16, n_events, n_channels, 128) each waveform split in 16 chuncks
+   # func -> (16, n_events, n_channels) pedestal for each chunck
+   if n_bins is not None:
+      baseline_values = func(np.split(wfs, 2048 // n_bins, axis=-1), axis=-1)
+
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, n_bins % 2048, axis=0)
+   else:
+      baseline_values = func(wfs, axis=-1)
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, 2048, axis=0)
+          
+   # np.moveaxis -> (n_events, n_channels, 2048)
+   baseline_traces = np.moveaxis(baseline_traces, 0, -1)
+    
+   return wfs - baseline_traces
+
+
+def get_time_offset(trigger_type):
+   """ 
+   Mapping the offset between trace start time and trigger time (~ signal time). 
+   Temporary use hard-coded values for each trigger type. In the future this
+   information might be time, station, and channel dependent and should come 
+   from a database (or is already calibrated in mattak)
+   
+   Parameters
+   ----------
+   
+   trigger_type: str
+      Trigger type encoded as string from Mattak
+      
+   Returns
+   -------
+   
+   time_offset: float
+      trace_start_time = trigger_time - time_offset 
+   
+   """
+   
+   time_offsets = {
+      "FORCE": 0,
+      "LT": 213 * units.ns,  # ~ 1 / 3 of trace @ 2048 sample with 3.2 GSa/s
+      "RADIANT": 320 * units.ns  # ~ 1 / 2 of trace @ 2048 sample with 3.2 GSa/s
+   }
+   
+   if trigger_type.startswith("RADIANT"):
+      trigger_type = "RADIANT"
+   
+   if trigger_type in time_offsets:
+      return time_offsets[trigger_type]
+   else:
+      raise KeyError(f"Unknown trigger type: {trigger_type}. Known are: FORCE, LT, RADIANT. Abort ....")
+
+
+class readRNOGData:
+
+   def begin(self, 
+             data_dirs, log_level=logging.INFO, selectors=None, 
+             read_calibrated_data=False,
+             apply_baseline_correction=True,
+             convert_to_voltage=True,
+             select_triggers=None,
+             select_runs=True,
```

Comment:
shouldn't the default be `False`? A frequent use case will be "I want to analyse run XX for station YY" ... i.e. the reader **should** work by just passing a file OR data dir (without any runtable)

Reply:
We can change the default to false. However, just to say it: If the run_table is not available an error is logged but the program is not stoped and no selection applied

Reply:
any thoughts ? @shallmann

Reply:
done.